login/crypto.c

```diff
@@ -44,10 +44,15 @@ static int login_tag(bool verify, const char* host_id, const char* action,
                      uint8_t output[GLOME_MAX_TAG_LENGTH]) {
   size_t message_len = strlen(host_id) + 1 + strlen(action) + 1;
   char* message = calloc(message_len, 1);
+  int ret;
   if (message == NULL) {
     return -1;
   }
-  int ret = snprintf(message, message_len, "%s/%s", host_id, action);
+  if (strlen(action) > 0) {
+    ret = snprintf(message, message_len, "%s/%s", host_id, action);
+  } else {
+    ret = snprintf(message, message_len, "%s", host_id);
```

Comment:
Uh-oh, I believe we have a specification problem here: is there a difference between an empty action and an absent action?

I'd rather not address this right here, if it's not needed for this PR.

Reply:
Sure, it is not used here directly.